File: Avaliacoes/Java/Modulo1/src/main/java/thread/CaixaThread.java
package thread;

import model.SaldoCentral;

/**
 * Representa um caixa físico do evento. Cada instância é uma thread que
 * vende uma quantidade fixa de fichas e soma o valor arrecadado ao
 * {@link SaldoCentral}, que é compartilhado com as outras 4 threads.
 *
 * @author Andrisa Santos
 */
public class CaixaThread extends Thread {

    /** Saldo compartilhado entre TODAS as threads de caixa. */
    private final SaldoCentral saldoCentral;

    /** Quantidade de fichas que este caixa deve vender. */
    private final int quantidadeFichas;

    /** Preço de cada ficha. */
    private final double precoFicha;

    /**
     * Cria uma nova thread de caixa.
     *
     * @param nome             nome da thread (ex.: "Caixa1")
     * @param saldoCentral     saldo compartilhado do evento
     * @param quantidadeFichas quantidade de fichas a vender
     * @param precoFicha       preço unitário da ficha
     */
    public CaixaThread(String nome, SaldoCentral saldoCentral, int quantidadeFichas, double precoFicha) {
        super(nome);
        this.saldoCentral = saldoCentral;
        this.quantidadeFichas = quantidadeFichas;
        this.precoFicha = precoFicha;
    }

    /**
     * Vende ficha por ficha, chamando o método sincronizado a cada venda
     */
    @Override
    public void run() {
        for (int i = 0; i < quantidadeFichas; i++) {
            saldoCentral.adicionarVenda(precoFicha);
        }
    }

    /** @return o saldo central associado a esta thread */
    public SaldoCentral getSaldoCentral() {
        return saldoCentral;
    }

    /** @return a quantidade de fichas que esta thread deve vender */
    public int getQuantidadeFichas() {
        return quantidadeFichas;
    }

    /** @return o preço de cada ficha vendida por esta thread */
    public double getPrecoFicha() {
        return precoFicha;
    }
}

File: Avaliacoes/CSharp/Modulo1/thread/CaixaThread.cs
using System.Threading;

/// <summary>
/// Representa um caixa físico do evento. Cada instância vende uma
/// quantidade fixa de fichas e soma o valor arrecadado ao
/// <see cref="SaldoCentral"/>, que é compartilhado com as outras 4 threads.
/// </summary>
public class CaixaThread
{
    private readonly Thread _thread;
    private readonly SaldoCentral _saldoCentral;
    private readonly int _quantidadeFichas;
    private readonly double _precoFicha;

    /// <summary>Nome do caixa (equivalente ao <c>getName()</c> do Java).</summary>
    public string Nome { get; }

    public CaixaThread(string nome, SaldoCentral saldoCentral, int quantidadeFichas, double precoFicha)
    {
        Nome = nome;
        _saldoCentral = saldoCentral;
        _quantidadeFichas = quantidadeFichas;
        _precoFicha = precoFicha;

        // O delegate "Executar" é o equivalente ao run() do Java: é o
        // código que efetivamente roda em uma thread separada.
        _thread = new Thread(Executar) { Name = nome };
    }

    /// <summary>
    /// Corpo da thread: simula a venda das fichas, uma a uma, chamando o
    /// método sincronizado do saldo central a cada venda.
    /// </summary>
    private void Executar()
    {
        for (int i = 0; i < _quantidadeFichas; i++)
        {
            _saldoCentral.AdicionarVenda(_precoFicha);
        }
    }

    /// <summary>Inicia a thread (equivalente a <c>start()</c> no Java).</summary>
    public void Start() => _thread.Start();

    /// <summary>Aguarda o término da thread (equivalente a <c>join()</c> no Java).</summary>
    public void Join() => _thread.Join();

    public SaldoCentral GetSaldoCentral() => _saldoCentral;
    public int GetQuantidadeFichas() => _quantidadeFichas;
    public double GetPrecoFicha() => _precoFicha;
}
